### PR TITLE
Add search adverts and improve UI

### PR DESCRIPTION
**Pull request by kickupduster**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,6 +1,7 @@
 class ApplicationController < ActionController::Base
   before_action :authenticate_user!
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_filter :set_search
```
Comment (iermakovets) [inline app/controllers/application_controller.rb]: use before_action, before filter is deprecated

File: app/helpers/categories_helper.rb [hunk 2]
```diff
@@ -1,2 +1,5 @@
 module CategoriesHelper
+  def category_options
+    Category.all.pluck(:name, :id)
```
Comment (iermakovets) [inline app/helpers/categories_helper.rb]: You can call method `pluck` on model: `Category.pluck(:name, :id)`

File: app/assets/stylesheets/application.css [hunk 3]
```diff
@@ -17,17 +17,30 @@
...
 }
-
-.panel
-{
+.search{
   margin-left: -230px;
   font-size: 17px;
   width: 200px;
   float: left;
+  position: absolute;
```
Comment (iermakovets) [inline app/assets/stylesheets/application.css]: I think this line is unnecessary. As I understood now you use flexbox

File: app/views/layouts/application.html.slim [hunk 4]
```diff
@@ -6,33 +6,38 @@ html
...
+                      span.caret
+                    ul.dropdown-menu role="menu"
+                      li = link_to "Edit user", edit_user_registration_path
+                      li = link_to "My profile", user_path(current_user.id)
+                      li.divider
+                      li = link_to "Log out", destroy_user_session_path, method: :delete
           - else
-            li = link_to "Sing up", new_user_registration_path
+            li
+              a.navbar-brand href="/users/sign_up" Sing up
```
Comment (iermakovets) [inline app/views/layouts/application.html.slim]: Return link_to

File: app/assets/stylesheets/application.css (hunk 3 above)
Comment (kickupduster) [inline app/assets/stylesheets/application.css]: I will fix it next pull request

**Comment by iermakovets** [comment]
🆗

**Pull request merged**